addons/binding/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/handler/BrokerHandler.java

```diff
@@ -179,40 +178,40 @@ protected void assignSSLContextProvider(BrokerHandlerConfig config, MqttBrokerCo
      */
     protected MqttBrokerConnection createBrokerConnection() throws IllegalArgumentException {
         String host = config.host;
-        if (StringUtils.isBlank(host) || host == null) {
+        if ((host == null) || StringUtils.isBlank(host)) {
             throw new IllegalArgumentException("Host is empty!");
         }
 
-        final MqttBrokerConnection connection = new MqttBrokerConnection(host, config.port, config.secure,
-                config.clientID);
+        MqttBrokerConnection connection = new MqttBrokerConnection(host, config.port, config.secure, config.clientID);
+        connection.setQos(config.qos);
+        connection.setRetain(config.retainMessages);
 
-        final String username = config.username;
-        final String password = config.password;
-        if (StringUtils.isNotBlank(username) && password != null) {
-            connection.setCredentials(username, password); // Empty passwords are allowed
+        if (StringUtils.isNotBlank(config.username) && (config.password != null)) {
```

Comment:
You can't use config properties directly. Otherwise our static null checker will warn. Makes sense actually. In a multi threaded environment another thread could alter `config.username` between the check and the usage.